src/FieldWarning/Assets/Loading/Worker.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Threading;
+using UnityEngine;
+
+public delegate IEnumerator WorkerCoroutineDelegate();
+public delegate void WorkerThreadDelegate();
+public delegate void WorkerConcurrentDelegate();
+
+public abstract class Worker
+{
+    public double PercentDone = 0;
+    public bool Finished = false;
+    public bool Started = false;
+    public string description;
+    
+    public Worker(string desc)
+    {
+        
+        description = desc;
+    }
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    protected void Begun()
+    {
+        Started = true;
+        PercentDone = 0;
+    }
+
+    protected void Ended()
```

Comment:
These aren't really function names. I agree that it's hard to name them but with the current names I had to look at the implementation to get what they're doing.

"SetProgressToStarted" and "SetProgressToFinished'?